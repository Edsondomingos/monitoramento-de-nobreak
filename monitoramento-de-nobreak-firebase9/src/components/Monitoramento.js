import * as React from 'react';
import { useState } from 'react';
import { Container, Titulo, TextoComum, Botao, Circulo, LigaDesliga, TextoBtn } from '../../assets/styledComponents/Components'
import { App} from './TesteMQTT'

export default function Monitoramento(props) {

  const [liga, setLigar] = useState(false);
  const [nome, setNome] = useState('Nome do Nobreak')
 
  
//{props.route.params.cod}
  return (
    <Container>
      
  
      <Titulo>{props.route.params.cod}</Titulo>

      <App />     

      
      <Botao
        onPress={() => props.navigation.navigate('EditarNobreak')}
      >
        <TextoBtn>Editar nobreak</TextoBtn>
      </Botao>
     
    </Container>
  );
}
