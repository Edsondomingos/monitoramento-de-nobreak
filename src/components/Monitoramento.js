import * as React from 'react';
import { useState } from 'react';
import { Container, Titulo, TextoComum, Botao, Circulo, LigaDesliga, TextoBtn } from '../../assets/styledComponents/Components'
import { App, Red, onoff } from './TesteMQTT'
// import Status from './statusEnergia'

export default function Monitoramento(props) {

  const [liga, setLigar] = useState(false);
  const [nome, setNome] = useState('Nome do Nobreak')
  const [nivel, setNivel] = useState(50)
 
  function switchLigar() {
    setLigar(!liga);
  }

  return (
    <Container>
      
  
      <Titulo>{nome}</Titulo>  
      
      <Circulo>
        <TextoComum>Uso de Energia</TextoComum>
        <Red />
        <TextoComum></TextoComum>
      </Circulo>
      
      
      <Circulo> 
        <TextoComum>Bateria</TextoComum>     
        <TextoComum><App /></TextoComum>
      </Circulo>

      <Circulo>
        <LigaDesliga 
            testID='btnLigar'
            style={liga ? {backgroundColor: '#7f7'} : {backgroundColor: '#f33'}}
            onPress={onoff}
          >
          {/*<TextoComum>{liga ? 'LIGADO' : 'DESLIGADO'}</TextoComum>*/}
          <TextoComum>BTN</TextoComum>
        </LigaDesliga>
      </Circulo>
      <Botao
        onPress={() => props.navigation.navigate('EditarNobreak')}
      >
        <TextoBtn>Editar nobreak</TextoBtn>
      </Botao>
     
    </Container>
  );
}
// onPress={onoff /*switchLigar*}