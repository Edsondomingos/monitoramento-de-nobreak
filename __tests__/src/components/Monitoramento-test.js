import renderer from 'react-test-renderer'
import { render, fireEvent } from "@testing-library/react-native";
import Monitoramento from '../../../src/components/Monitoramento'
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe('Teste de snapshot', () => {
    it('Snapshot tela Monitoramento', () => {
        const tela = renderer.create(<Monitoramento />).toJSON()
        expect(tela).toMatchSnapshot()
    })
    
})

describe("Teste de inicialização e Exibição", () => {
    it("Components tela Monitoramento", () => {
        const { getByText, getByTestId} = render(<Monitoramento />);
        
        expect(getByText("Uso de Energia")).toBeTruthy();
        expect(getByText("Bateria")).toBeTruthy();

        const btnLigar = getByTestId('btnLigar')
        fireEvent.press(btnLigar)
        expect(getByText('LIGADO'))
    })
})