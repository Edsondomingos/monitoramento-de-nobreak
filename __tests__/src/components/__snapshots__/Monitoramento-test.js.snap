// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot Snapshot tela Monitoramento 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 30,
          "textAlign": "center",
        },
      ]
    }
  >
    Nome do Nobreak
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#000",
          "borderRadius": 100,
          "borderStyle": "solid",
          "borderWidth": 5,
          "display": "flex",
          "height": 180,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
          "width": 180,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 22,
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      Uso de Energia
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#000",
          "borderRadius": 100,
          "borderStyle": "solid",
          "borderWidth": 5,
          "display": "flex",
          "height": 180,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
          "width": 180,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 22,
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      Bateria
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 22,
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      50
      %
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#000",
          "borderRadius": 100,
          "borderStyle": "solid",
          "borderWidth": 5,
          "display": "flex",
          "height": 180,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
          "width": 180,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#7f7",
            "borderRadius": 100,
            "borderWidth": 1,
            "display": "flex",
            "height": 150,
            "justifyContent": "center",
            "marginBottom": "auto",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "auto",
            "width": 150,
          },
          Object {
            "backgroundColor": "#f33",
          },
        ]
      }
      testID="btnLigar"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 22,
              "marginBottom": 10,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        DESLIGADO
      </Text>
    </View>
  </View>
</View>
`;
